QappTuning.py

```diff
@@ -0,0 +1,585 @@
+    else:
+        full_candidates.sort(key=sortBitDepth)
+        mode = full_candidates[-1]
+main = {"size": (800, 600)}
+sensor = {'output_size': mode['size'], 'bit_depth': mode['bit_depth']}
+config = picam2.create_preview_configuration(main=main, sensor=sensor)
+picam2.configure(config)
+
+
+# Function that returns true if it can detect macbeth plot
+def check(adress, target):
+    Cam = Camera("imx3.json", json=json_template)
+    Img = dng_load_image(Cam, adress)
+
+    av_chan = (np.mean(np.array(Img.channels), axis=0) / (2 ** 16))
+    av_val = np.mean(av_chan)
+
+    if av_val < Img.blacklevel_16 / (2**16) + 1 / 64:
+        return False
+    else:
+        macbeth = find_macbeth(Cam, av_chan, mac_config=(0, 0))
+    if macbeth is None:
+        return False
+    else:
+        return True
+
+
+# add camera name to this folder that is default
+index_photo = 0
+start = False
+for f in os.listdir(os.path.expanduser("~")):
+    if picam2.global_camera_info()[0]["Model"] in f:
+        index_photo += 1
+if index_photo == 0:
+    folder_directory = os.path.join(os.path.expanduser("~"), picam2.global_camera_info()[0]["Model"])
+else:
+    folder_directory = os.path.join(os.path.expanduser("~"), picam2.global_camera_info()[0]["Model"] + "_" + str(index_photo))
+
+
+# what pi are we working on?
+if Picamera2.platform == Platform.PISP:
+    target = "pi5"
+    from ctt_pisp import json_template
+else:
+    target = "pi4"
+    from ctt_vc4 import json_template
+
+
+# checks if sting is a valid path to a folder
+def valid_path(s):
+    if (s[0] != "/" or s[-1] == "/"):
+        return False
+    else:
+        return True
+
+
+# create the screen that shades if it can se a macbeth plote clos to camera
+# macbeth reference data
+Cam = Camera("imx3.json", json=json_template)
+ref = cv2.imread(Cam.path + 'ctt_ref.pgm', flags=cv2.IMREAD_GRAYSCALE)
+ref_w = 120
+ref_h = 80
+rc1 = (0, 0)
+rc2 = (0, ref_h)
+rc3 = (ref_w, ref_h)
+rc4 = (ref_w, 0)
+ref_corns = np.array((rc1, rc2, rc3, rc4), np.float32)
+ref_data = (ref, ref_w, ref_h, ref_corns)
+
+# function that checks for macbeth close to camera
+
+
+def my_find_macbeth(img):
+    warnings.simplefilter("ignore")
+    fxn()
+    cor, mac, coords, msg = get_macbeth_chart(img, ref_data)
+    if cor < 0.75:
+        a = 2
+        img_br = cv2.convertScaleAbs(img, alpha=a, beta=0)
+        cor_b, mac_b, coords_b, msg_b = get_macbeth_chart(img_br, ref_data)
+        if cor_b > cor:
+            cor, _, coords, _ = cor_b, mac_b, coords_b, msg_b
+    if cor < 0.75:
+        a = 4
+        img_br = cv2.convertScaleAbs(img, alpha=a, beta=0)
+        cor_b, mac_b, coords_b, msg_b = get_macbeth_chart(img_br, ref_data)
+        if cor_b > cor:
+            cor, _, coords, _ = cor_b, mac_b, coords_b, msg_b
+    if cor > 0.5:
+        return (cor, coords)
+    else:
+        return (0, None)
+
+# future parameters
+
+
+pool = ThreadPoolExecutor(max_workers=4)
+overlay_active = False
+busy = False
+
```

Comment:
Do you think `busy` needs to be a global variable, or would some kind of `self.busy` work instead?

Reply:
I think it needs to be global because it is used in two different classes. Both MacbethWindow and Widgte one.